compose_generator/odcs_fetcher.py

```diff
@@ -17,5 +21,21 @@ class ODCSFetcher(ComposeFetcher):
     """
     Fetch ODCS compose based on a remote compose-reference and store it locally
     """
-    def __call__(self, request_reference: ComposeReference) -> ODCSResultReference:
-        raise NotImplementedError()
+
+    def __call__(
+        self, request_reference: ComposeReference, download_path: Optional[Path] = None
```

Comment:
I'm not sure if we can add another input parameter here and still comply with the protocol. I think the path can be a member of the class instead.

Reply:
I don't think we should add it to the generic implementation nor I'm sure it is even needed. any reason not to tmpdir it, with the current foreseeable usage??

Reply:
I'm good with that.

Reply:
Ended up using it, wanted to avoid monkeypatching a path